php-frontend/src/main/java/org/sonar/php/parser/TreeFactory.java

```diff
@@ -1363,10 +1365,12 @@ public SpreadArgumentTree spreadArgument(InternalSyntaxToken ellipsis, Expressio
 
   public FunctionCallTree functionCallParameterList(
     InternalSyntaxToken openParenthesis,
-    SeparatedListImpl<ExpressionTree> arguments,
+    SeparatedListImpl<FunctionCallArgumentTree> arguments,
     InternalSyntaxToken closeParenthesis
   ) {
-    return new FunctionCallTreeImpl(openParenthesis, arguments, closeParenthesis);
+    return new FunctionCallTreeImpl(openParenthesis,
+      new SeparatedListImpl<>(arguments, arguments.getSeparators()),
```

Comment:
Why do we create a new `SeparatedListImpl` instead of just using `arguments`?

Reply:
There is no reason. Thanks for noticing! I think I introduced the `new SeparatedListImpl` in an earlier [commit](https://github.com/SonarSource/sonar-php/pull/679/commits/2dd8ca550d6636c0b17b1928dc9d1c2c0559f01f#diff-05472a9f22d1bf00b87d87cf602878d8R1376) where I only wanted to having the parsing work, and forgot to change it back correctly :)